wallet/walletrpc.c

```diff
@@ -177,29 +177,54 @@ static const struct json_command newaddr_command = {
 };
 AUTODATA(json_command, &newaddr_command);
 
-static struct command_result *json_listaddrs(struct command *cmd,
+static void json_add_addr(struct json_stream *response,
+				 const u64 keyidx,
+				 const struct pubkey *pubkey,
+				 const char *out_p2wpkh,
+				 const u8 *redeemscript_p2wpkh,
+				 const char *out_p2tr)
+{
+	json_object_start(response, NULL);
+	json_add_u64(response, "keyidx", keyidx);
+	json_add_pubkey(response, "pubkey", pubkey);
+	json_add_string(response, "bech32", out_p2wpkh);
+	json_add_hex_talarr(response, "bech32_redeemscript", redeemscript_p2wpkh);
+	json_add_string(response, "p2tr", out_p2tr);
```

Comment:
Always returning the P2WPKH and P2TR for every key index is unfortunate but can't be avoided with the current architecture and without storing more information in the database when addresses are created

Reply:
... which we should definitely do!

Reply:
Reused @rustyrussell's  code from `remember-keytype` branch to get issued `addrtype` from DB. Ready for review again.